File: UnitC/ch06/data-6.5-swimming-suit.jl


## 1. load package
include("../../utils.jl")
using UnicodePlots

## 2. data processing
    desc=Lock5Table(502,"Wetsuits","more fast with wetsuit?",["Wetsuit", "NoWetsuit", "Gender", "Type"])
    df= @pipe load_csv(desc.name,false)|>select(_,desc.feature[1:2])|>permutedims
    #@pt df 
#= 
┌─────────┬─────────┬─────────┬─────────┬─────────┬─────────┬─────────┬─────────┬─────────┬─────────┬─────────┬─────────┐
│      x1 │      x2 │      x3 │      x4 │      x5 │      x6 │      x7 │      x8 │      x9 │     x10 │     x11 │     x12 │
│ Float64 │ Float64 │ Float64 │ Float64 │ Float64 │ Float64 │ Float64 │ Float64 │ Float64 │ Float64 │ Float64 │ Float64 │
├─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
│    1.57 │    1.47 │    1.42 │    1.35 │    1.22 │    1.75 │    1.64 │    1.57 │    1.56 │    1.53 │    1.49 │    1.51 │
│    1.49 │    1.37 │    1.35 │    1.27 │    1.12 │    1.64 │    1.59 │    1.52 │     1.5 │    1.45 │    1.44 │    1.41 │
└─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┘
=#

   data = [df[1,:]|>Vector,df[2,:]|>Vector]
   cats=["wetsuits","no-wetsuits"]
   summary_df = DataFrame(Group=cats,
    n=size.(data, 1),
    Mean=mean.(data),
    Stddev=std.(data)
    )
    @pt summary_df    
#= 
    ┌─────────────┬───────┬─────────┬──────────┐
    │       Group │     n │    Mean │   Stddev │
    │      String │ Int64 │ Float64 │  Float64 │
    ├─────────────┼───────┼─────────┼──────────┤
    │    wetsuits │    12 │ 1.50667 │ 0.136271 │
    │ no-wetsuits │    12 │ 1.42917 │ 0.141065 │
    └─────────────┴───────┴─────────┴──────────┘
=#

## 4. box plot
UnicodePlots.boxplot(cats, data, title="different speed", xlabel="speed", ylabel="group")
#= 
   different speed              
                     ┌                                        ┐ 
                            ╷            ┌───┬──┐         ╷     
            wetsuits        ├────────────┤   │  ├─────────┤     
   group                    ╵            └───┴──┘         ╵     
                      ╷             ┌────┬──┐       ╷           
         no-wetsuits  ├─────────────┤    │  ├───────┤           
                      ╵             └────┴──┘       ╵           
                     └                                        ┘ 
                      1.1              1.45                1.8  
                                        speed      
=#

File: marimo/ch06/data-6.5-swimming-suit.py
import marimo

__generated_with = "0.24.0"
app = marimo.App(width="medium")


@app.cell
def _():
    import marimo as mo
    import sys
    from pathlib import Path
    import numpy as np
    import polars as pl
    import matplotlib.pyplot as plt
    from scipy import stats
    from great_tables import GT
    _p = Path(mo.notebook_location()).resolve().parent
    while _p != _p.parent and not (_p / "lock5.py").exists():
        _p = _p.parent
    if str(_p) not in sys.path:
        sys.path.insert(0, str(_p))
    import lock5
    return mo, sys, Path, np, pl, plt, stats, GT, lock5


@app.cell
def _(mo):
    mo.md(r"""# Data 6.5 — Are wetsuits faster? (paired swim speeds)

Port of `UnitC/ch06/data-6.5-swimming-suit.jl`: compare the paired `Wetsuit` and
`NoWetsuit` swim speeds with a summary table and a grouped box plot.""")
    return


@app.cell
def _(lock5, np):
    desc = lock5.Lock5Table(502, "Wetsuits", "more fast with wetsuit?", ["Wetsuit", "NoWetsuit", "Gender", "Type"])
    df = lock5.load_csv(desc.name, drop=False).select(desc.feature[:2])
    wet = df["Wetsuit"].to_numpy().astype(float)
    no_wet = df["NoWetsuit"].to_numpy().astype(float)
    data = [wet, no_wet]
    cats = ["wetsuits", "no-wetsuits"]
    summary = lock5.summary_df(cats, data)
    return desc, df, wet, no_wet, data, cats, summary


@app.cell
def _(GT, mo, summary):
    mo.Html(GT(summary).tab_header(title="Swim speed by suit").fmt_number(columns=["Mean", "Stddev"], decimals=4).as_raw_html())
    return


@app.cell
def _(cats, data, lock5):
    fig, ax = lock5.grouped_boxplot(cats, data, title="different speed", xlabel="speed", ylabel="group")
    fig
    return fig, ax


if __name__ == "__main__":
    app.run()
